src/core/net/mdns.hpp

```diff
@@ -552,6 +563,71 @@ class Core : public InstanceLocator, private NonCopyable
      */
     void SetMaxMessageSize(uint16_t aMaxSize) { mMaxMessageSize = aMaxSize; }
 
+    /**
+     * Allocates a new iterator.
+     *
+     * @returns A pointer to the newly allocated iterator or `nullptr` if fails to allocate.
+     *
+     */
+    Iterator *AllocateIterator(void);
+
+    /**
+     * Frees a previously allocated iterator.
+     *
+     * @param[in] aIterator  The iterator to free.
+     *
+     */
+    void FreeIterator(Iterator &aIterator);
+
+    /**
+     * Iterates over registered host entries.
+     *
+     * On success, @p aHost is populated with the next host information. Pointers within the `Host` structure
+     * (like `mName`) remain valid until the next call to any OpenThread stack public or platform API/callback.
+     *
+     * @param[in]  aIterator   The iterator to use.
+     * @param[out] aHost       A `Host` to return the information about next host entry.
+     * @param[out] aState      An `EntryState` to return the entry state.
+     *
+     * @retval kErrorNone         @p aHost, @p aState, & @p aIterator are updated successfully.
+     * @retval kErrorNotFound     Reached the end of the list.
+     * @retval kErrorInvalidArg   Iterator is not valid.
+     *
+     */
+    Error GetNextHost(Iterator &aIterator, Host &aHost, EntryState &aState) const;
+
+    /**
+     * Iterates over registered service entries.
+     *
+     * On success, @p aService is populated with the next service information. Pointers within the `Service` structure
+     * (like `mServiceType`) remain valid until the next call to any OpenThread stack public or platform API/callback.
+     *
+     * @param[out] aService    A `Service` to return the information about next service entry.
+     * @param[out] aState      An `EntryState` to return the entry state.
+     *
```

Comment:
```suggestion
     * @param[out] aState      An `EntryState` call to return the entry state.
     *
```

Reply:
if you agree with the suggested change, you could make the same type of change to other instances parameter references.